src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -65,20 +68,25 @@ export class PeoplePicker extends React.PureComponent<IPeoplePickerProps, IPeopl
 
     @autobind
     private _renderSuggestions(): JSX.Element {
+        let allSelected: boolean = true;
         return (
             <div className="people-picker-suggestions">
                 {this.props.loadingSuggestionList && <Spinner type={SpinnerType.small} />}
-                {!this.props.loadingSuggestionList && this.props.suggestionList.map((principal, index) => {
+                {!this.props.loadingSuggestionList && this.props.suggestionList.length > 0 && this.props.suggestionList.map((principal, index) => {
                     const alreadySelected = this.state.selectedPrincipalList 
                         && this.state.selectedPrincipalList.find(selected => selected.identifier === principal.identifier) !== undefined;
 
+                    allSelected = allSelected && alreadySelected;    
                     return !alreadySelected && <Principal
                         key={principal.identifier}
                         principal={principal}
                         isSelected={false}
                         onSelect={this._onSuggestionClick}
                     />;
                 })}
+                {!this.props.loadingSuggestionList && (this.props.suggestionList.length === 0 || allSelected) && <div className="no-result">
```

Comment:
why are we showing no results text when allSelected? for that matter what is the purpose of allSelected, it starts as set to true and thats it, or am i missing something

Reply:
@vinkobedek this is just a check if in the suggestion list are only principals that are all already selected, because we are not showing principals that are already selected so this situation is the same as if a user typed a term that returns 0 principals